### PR TITLE
pull in transfer limit code from HDFS (SOFTWARE-2512)

### PR DESCRIPTION
**Pull request by edquist**
This is largely taken from parts of gridftp_hdfs.c, with most of the hdfs-specific stuff stripped out.

### CODE REVIEW

File: src/osg_extension_dsi.c [hunk 1]
```diff
@@ -149,9 +155,199 @@ osg_extensions_init(globus_gfs_operation_t op, globus_gfs_session_info_t * sessi
...
+    // only used for Error macros
+    char local_host[256] = {};
+    if (gethostname(local_host, 255)) {
+        strcpy(local_host, "UNKNOWN");
+    }
+
+    int user_lock_count = 0;
+    if (user_transfer_limit > 0) {
+        char user_sem_name[256];
+        snprintf(user_sem_name, 255, "/dev/shm/gridftp-hdfs-%s-%d", username, user_transfer_limit);
```
Comment (edquist) [inline src/osg_extension_dsi.c]: Do we want to keep `gridftp-hdfs` in the sem name?

Comment (bbockelm) [reply src/osg_extension_dsi.c]: Nope!

**Review by bbockelm — APPROVED** [review]
I pushed a few fixes to the branch to further scrub out HDFS-isms.

**Comment by bbockelm** [comment]
@edquist - how much testing did you do for this?  Sufficient to tag and push to other testers?

**Comment by edquist** [comment]
I forget actually, it's been a while.  It was probably only light "can I still transfer a file?" testing.

**Comment by bbockelm** [comment]
Ah - there's no update to the README!  Can you include that?  Just a few lines.

I'm taking care of the developer testing.

**Comment by bbockelm** [comment]
Ok, I'm happy now!

Actually had to make it fail when the limits are hit, but otherwise minor stuff.

**Pull request merged**